src/main/scala/org/codeoverflow/chatoverflow/requirement/service/rcon/RconConnector.scala

```diff
@@ -0,0 +1,129 @@
+package org.codeoverflow.chatoverflow.requirement.service.rcon
+
+import java.io.{DataInputStream, InputStream, OutputStream}
+import java.net.{Socket, SocketException}
+import java.nio.{ByteBuffer, ByteOrder}
+import java.util.Random
+
+import org.codeoverflow.chatoverflow.WithLogger
+import org.codeoverflow.chatoverflow.connector.Connector
+
+class RconConnector(override val sourceIdentifier: String) extends Connector(sourceIdentifier) with WithLogger {
+  override protected var requiredCredentialKeys: List[String] = List("password", "address")
+  override protected var optionalCredentialKeys: List[String] = List("port")
+
+  private var socket: Socket = _
+  private var outputStream: OutputStream = _
+  private var inputStream: InputStream = _
+  private var requestId: Int = 0
+  private var loggedIn = false
+
+  def sendCommand(command: String): String = {
+    if (!loggedIn) {
+      logger error "Could not execute RCON Command due to wrong password or no connection"
+      return null
+    }
+    logger debug s"Sending $command to RCON"
+    requestId += 1
+    if (write(2, command.getBytes("ASCII"))) {
+      return read()
+    }
+    null
+  }
+
+
+  /**
+    * Starts the connector, e.g. creates a connection with its platform.
+    */
+  override def start(): Boolean = {
+    logger info s"Starting rcon connection to ${credentials.get.getValue("address").get}"
+    var port: Int = 25575
+    if (credentials.get.exists("port")) {
+      port = credentials.get.getValue("port").get.toInt
+      if (port < 1 || port > 65535) {
+        return false
```

Comment:
scala does not have a return statement. this would not leave the function,

Reply:
Well, scala does have a return statement, which you can use for early returns: https://scastie.scala-lang.org/B9Ryg8hlRUWnQ8hEWHwIsA. When you want to return a value and the statement is in the last line of e.g. a if else block or a method you can omit the `return` keyword. Thats why you don't see it that much.

Reply:
Just tested: Looks like it works.

If the port is not valid it will quit starting at that point.

Reply:
Wow... I was wrong about this for years then. I thought it would go in the same direction like removing break and continue for loops...